proxy/director.go

```diff
@@ -58,6 +64,10 @@ func (d *director) refresh() {
 			log.Printf("proxy: upstream URL invalid: %v", err)
 			continue
 		}
+		host, port, _ := net.SplitHostPort(uu.Host)
+		if _, ok := d.banned[host+":"+port]; ok {
+			continue
```

Comment:
when used with `port` and `proxy`, this is currently useless.
I am waiting on code review for better ways to update `endpoint`.